migration_scripts/convert_parties.py

```diff
@@ -0,0 +1,301 @@
+# If you have extra members in 'TrainerMon':
+# 1. Add a regular expression which matches that member (e.g. 'shadow_definition').
+# 2. Match that regular expression in 'convert' and write into 'attributes' with the key that 'trainerproc' should parse.
+# 3. Add the key used in 'attributes' to 'pokemon_attribute_order'.
+# 4. Update 'trainerproc.c' to parse the new key.
+
+import re
+import sys
+
+is_blank = re.compile(r'^[ \t]*(//.*)?$')
+
+begin_party_definition = re.compile(r'struct TrainerMon (\w+)\[\] =')
+end_party_definition = re.compile(r'^};')
+begin_pokemon_definition = re.compile(r'^    { *$')
+end_pokemon_definition = re.compile(r'^    },? *$')
+level_definition = re.compile(r'\.lvl = (\d+)')
+species_definition = re.compile(r'\.species = SPECIES_(\w+)')
+gender_definition = re.compile(r'\.gender = TRAINER_MON_(\w+)')
+nickname_definition = re.compile(r'\.nickname = COMPOUND_STRING\("([^"]+)"\)')
+item_definition = re.compile(r'\.heldItem = ITEM_(\w+)')
+ball_definition = re.compile(r'\.ball = ITEM_(\w+)')
+ability_definition = re.compile(r'\.ability = ABILITY_(\w+)')
+friendship_definition = re.compile(r'\.friendship = (\d+)')
+shiny_definition = re.compile(r'\.isShiny = (\w+)')
+ivs_definition = re.compile(r'\.iv = TRAINER_PARTY_IVS\(([0-9 ]+),([0-9 ]+),([0-9 ]+),([0-9 ]+),([0-9 ]+),([0-9 ]+)\)')
+evs_definition = re.compile(r'\.ev = TRAINER_PARTY_EVS\(([0-9 ]+),([0-9 ]+),([0-9 ]+),([0-9 ]+),([0-9 ]+),([0-9 ]+)\)')
+moves_definition = re.compile(r'\.moves = \{([^}]+)\}')
+move_definition = re.compile(r'MOVE_(\w+)')
+nature_definition = re.compile(r'\.nature = NATURE_(\w+)')
+
+species_replacements = {
+    "HO_OH": "Ho-Oh",
+    "PORYGON_Z": "Porygon-Z",
+    "TYPE_NULL": "Type: Null",
+    "JANGMO_O": "Jangmo-o",
+    "HAKAMO_O": "Hakamo-o",
+    "KOMMO_O": "Kommo-o",
```

Comment:
Totem Kommo-O handling

Reply:
I think this one already works. The replacements work via substrings, so I think we'd get `KOMMO_O_TOTEM` ⇒ `Kommo-o_TOTEM` ⇒ `Kommo-o-Totem`.